src/python_testing/TC_OCC_2_1.py

```diff
@@ -0,0 +1,253 @@
+        cluster = Clusters.Objects.OccupancySensing
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_OCC_2_1(self) -> str:
+        return "[TC-OCC-2.1] Attributes with DUT as Server"
+
+    def steps_TC_OCC_2_1(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Read Occupancy attribute."),
+            TestStep(3, "Read OccupancySensorType attribute."),
+            TestStep(4, "Read OccupancySensorTypeBitmap attribute."),
+            TestStep(5, "Read HoldTimeLimits attribute, if supported"),
+            TestStep(6, "Read HoldTime attribute, if supported"),
+            TestStep(7, "Read PIROccupiedToUnoccupiedDelay attribute, if supported"),
+            TestStep(8, "Read PIRUnoccupiedToOccupiedDelay attribute, if supported"),
+            TestStep(9, "Read PIRUnoccupiedToOccupiedThreshold attribute, if supported"),
+            TestStep(10, "Read UltrasonicOccupiedToUnoccupiedDelay attribute, if supported"),
+            TestStep(11, "Read UltrasonicUnoccupiedToOccupiedDelay attribute, if supported"),
+            TestStep(12, "Read UltrasonicUnoccupiedToOccupiedThreshold attribute, if supported"),
+            TestStep(13, "Read PhysicalContactOccupiedToUnoccupiedDelay attribute, if supported"),
+            TestStep(14, "Read PhysicalContactUnoccupiedToOccupiedDelay attribute, if supported"),
+            TestStep(15, "Read PhysicalContactUnoccupiedToOccupiedThreshold attribute, if supported")
+        ]
+        return steps
+
+    def pics_TC_OCC_2_1(self) -> list[str]:
+        pics = [
+            "OCC.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_OCC_2_1(self):
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        self.step(1)
+        attributes = Clusters.OccupancySensing.Attributes
+        attribute_list = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.AttributeList)
+
+        self.step(2)
+        if attributes.Occupancy.attribute_id in attribute_list:
+            occupancy_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.Occupancy)
+            asserts.assert_less_equal(occupancy_dut, 0b00000001, "Occupancy attribute is not in valid range")
+        else:
+            logging.info("Occupancy attribute is a mandatory attribute. Test step fails.")
+            asserts.fail("Missing mandatory attribute Occupancy")
+
+        self.step(3)
+        if attributes.OccupancySensorType.attribute_id in attribute_list:
+            occupancy_sensor_type_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.OccupancySensorType)
+            asserts.assert_less(occupancy_sensor_type_dut, Clusters.Objects.OccupancySensing.Enums.OccupancySensorTypeEnum.kUnknownEnumValue,

```

Comment:
I don't think you should allow the test to rely on actual constant value of kUnknown. Test plan lists what values are valid. This sounds like a:

```py
asserts.assert_in(
   occupancy_sensor_type_dut,
   {
       Clusters.Objects.OccupancySensing.Enums.OccupancySensorTypeEnum.kPIR,
       Clusters.Objects.OccupancySensing.Enums.OccupancySensorTypeEnum.kUltrasonic,
       # .... (code not tested ... please fill in correctly)
   },
   "OccupancySensorType is not in valid range"
)

Reply:
Done.